block/manager.go

```diff
@@ -564,42 +665,47 @@ func (m *Manager) trySyncNextBlock(ctx context.Context, daHeight uint64) error {
 		default:
 		}
 		currentHeight := m.store.Height()
-		b, ok := m.blockCache.getBlock(currentHeight + 1)
+		h, ok := m.headerCache.getHeader(currentHeight + 1)
+		if !ok {
+			m.logger.Debug("header not found in cache", "height", currentHeight+1)
+			return nil
+		}
+		d, ok := m.dataCache.getData(currentHeight + 1)
 		if !ok {
-			m.logger.Debug("block not found in cache", "height", currentHeight+1)
+			m.logger.Debug("data not found in cache", "height", currentHeight+1)
```

Comment:
So I think my comment on the data cache is highlighted here. 
If there was an empty block we would get stuck here I believe because we don't store `nil` data. 
Unless `nil` data doesn't mean empty block.

Reply:
Empty block (`Data`) is not `nil`, because there's always a metadata IIUC.